lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -175,6 +175,12 @@ namespace Js
             }
 
             OUTPUT_TRACE_DEBUGONLY(Js::ModulePhase, _u("\t>NotifyParentAsNeeded\n"));
+            if (this->promise != nullptr)
+            {
+                SourceTextModuleRecord::ResolveOrRejectDynamicImportPromise(false, this->GetErrorObject(), this->GetScriptContext(), this);
+                this->SetPromise(nullptr);
```

Comment:
In fact ResolveOrRejectDynamicImportPromise() takes care of this already. Removing this line and in similar cases

---
In reply to: [115382160](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115382160) [](ancestors = 115382160)